specs/kiota.config.md

```diff
@@ -0,0 +1,634 @@
+# Kiota Config
+
+Kiota generates client code for an API and stores parameters in a kiota.lock file. A project can contain multiple API clients, but they are independently managed. Kiota has no awareness that an app has a dependency on multiple APIs, even though that is a core use case.
+
+## Status
+
+| Date | Version | Author | Status |
+| -- | -- | -- | -- |
+| November 30th, 2023 | v0.3 | Sébastien Levert | Final Draft |
+| November 22nd, 2023 | v0.2 | Sébastien Levert | Draft |
+| September 24th, 2023 | v0.1 | Darrel Miller | Draft |
+
+## Current Challenges
+
+- Client code generation is not reproducible if API description changes
+- Kiota doesn’t have a good solution for APIs that use multiple security schemes.
+- Kiota doesn’t provide any support for generating auth providers with the required permissions, partially because currently we generate one client for APIs that use different schemes. How would we know which auth provider to generate.
+- Kiota doesn’t have a good story for acquiring a client identifier. e.g. apikey or OAuth2 ClientId. This could be possible if the OpenIDConnect URL pointed to a dynamic registration endpoint.
```

Comment:
to me this point and the previous are effectively the same thing, I'm not sure why you needed to include 2?